src/transaction-runner.coffee

```diff
@@ -359,17 +362,18 @@ class TransactionRunner
         #runAfterHooks
         @runHooksForData hooks.afterAllHooks, transactions, true, callback
 
-  executeTransaction: (transaction, hooks, callback) =>
-    unless callback
-      callback = hooks
-      hooks = null
-
-    configuration = @configuration
-
+  getRequestOptionsFromTransaction: (transaction) ->
+    requestOptions =
+      host: transaction.host
+      port: transaction.port
+      path: transaction.fullPath
+      method: transaction.request['method']
+      headers: transaction.request.headers
 
-    # Add length of body if no Content-Length present
-    # Doing here instead of in configureTransaction, because request body can be edited in before hook
+    return requestOptions
 
+  # Add length of body if no Content-Length present
+  setContentLength: (transaction) ->
```

Comment:
This function could be simplified by [caseless](https://www.npmjs.com/package/caseless), I believe.